src/smart-components/workspaces/WorkspaceListTable.tsx

```diff
@@ -107,6 +118,47 @@ const WorkspaceListTable = () => {
     setIsDeleteModalOpen(!isDeleteModalOpen);
   };
 
+  const isValidEditType = (workspace: Workspace) => {
+    switch (workspace.type) {
+      case 'root':
+        return false;
+      case 'default':
+        return true;
+      case 'ungrouped-hosts':
+        return false;
+      case 'standard':
+        return true;
+    }
+  };
+
+  const isValidDeleteType = (workspace: Workspace) => {
+    switch (workspace.type) {
+      case 'root':
+        return false;
+      case 'default':
+        return false;
+      case 'ungrouped-hosts':
+        return false;
+      case 'standard':
+        return true;
+    }
+  };
+
+  const canModify = (workspace: Workspace, action: 'edit' | 'delete') => {
+    if (
+      userPermissions.permission === 'inventory:groups:write' &&
+      (userPermissions.resourceDefinitions.length === 0 || userPermissions.resourceDefinitions.includes(workspace.id))
```

Comment:
The full shape of Resouce definition is as:
```
[{
    "resourceDefinitions":[
      {"attributeFilter": {"key":"group.id", "value": string[], "operation":"in"} }
    ],
    "permission":"inventory:groups:write"
}]
```

We are looking for the `value`
```suggestion
      (userPermissions.resourceDefinitions.length === 0 || userPermissions.resourceDefinitions.some((item) => item.attributeFilter.value.includes(workspace.id))
```

We could add a bit more safe guards here since the operation can be different and key as well. But that applies only for RDs created by API.